hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
Sorry, I started twitching and can't read anymore.

On Mon, Jun 13, 2016 at 10:50 AM, Steve Kuznetsov <notifications@github.com>
wrote:

> In hack/test-go.sh
> <https://github.com/openshift/origin/pull/5274#discussion_r66805178>:
>
> > @@ -77,22 +77,45 @@ fi
> >
> >  KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
> >
> > -if [ "${1-}" != "" ]; then
> > -  if [[ "${1}" == *"/..." ]]; then
> > +# Declare and set arrays to null string. We need to always keep the first
> > +# element to be the null string, otherwise the variable is considered unset.
> > +# When using, slice it to ignore the first element: ${varname[@]:1}.
> > +declare -a test_flags= test_packages=
>
> In case anyone's still watching this ... the *real* correct way to do
> this which
>
>    - correctly passes ('') as an array of one element, the empty string
>    - correctly passes () as an empty array
>    - doesn't split the array members on spaces
>
> $ my_function "${array[@]+"${array[@]}"}"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/5274/files/dcdeaf69876b254b961d9627207e5a4cb55da6a7#r66805178>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3zomZUH3-cwAfdwF0UMJqQ2KzBPks5qLW49gaJpZM4GS4wI>
> .
>